### PR TITLE
Implement simple application

### PR DESCRIPTION
**Pull request by wdebsqi**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from sklearn.feature_extraction.text import CountVectorizer
+
+class Application:
+    def __init__(self):
+        self.source_data = []
+        self.import_model_vocabulary()
+        self.preprocessor = pp.Preprocessor()
+        self.vectorizer = CountVectorizer(vocabulary=self.vocabulary)
+
+    def import_model_vocabulary(self):
```
Comment (Bilski311) [inline app.py:13]: I would split this into 2 separate methods.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        with open('model.pickle', 'rb') as file:
+            self.model = pickle.load(file)
+            
+        with open('vocabulary.pickle', 'rb') as file:
+            self.vocabulary = pickle.load(file)
+
+    def get_user_input(self):
+        self.source_data = []
```
Comment (Bilski311) [inline app.py:21]: This is initialized in constructor of this class and therefore this declaration is redundant.

File: app.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import preprocessor as pp
+import pandas as pd
+import pickle
+from sklearn.feature_extraction.text import CountVectorizer
+
+class Application:
+    def __init__(self):
+        self.source_data = []
```
Comment (Bilski311) [inline app.py:8]:
```suggestion
        self.user_inputs = []
```

File: app.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    def import_model_vocabulary(self):
+        with open('model.pickle', 'rb') as file:
+            self.model = pickle.load(file)
+            
+        with open('vocabulary.pickle', 'rb') as file:
+            self.vocabulary = pickle.load(file)
+
+    def get_user_input(self):
+        self.source_data = []
+        self.source_data.append(input('Put your text here: '))
```
Comment (Bilski311) [inline app.py:22]:
```suggestion
        user_input = input('Put your text here: ')
        self.user_inputs.append(user_input)
```

File: app.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    def print_menu(self):
+        print(50 * '-')
+        print('Welcome to spam recognizer! Type in any text and let the model decide whether its SPAM or HAM')
+        print('To exit, just type exit()')
+        print(50 * '-')
+    
+    def normalize_user_input(self):
+        self.user_input_df = pd.DataFrame(self.source_data, columns=['message'])
```
Comment (Bilski311) [inline app.py:31]:
```suggestion
        self.user_input_df = pd.DataFrame(self.user_inputs, columns=['message'])
```

File: app.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        self.preprocessed_data = self.preprocessor.clean_column_containing_text(self.user_input_df, 'message')
+        self.vectorized = self.vectorizer.transform(self.preprocessed_data['message'])
```
Comment (Bilski311) [inline app.py:33]:
```suggestion
        self.vectorized_inputs = self.vectorizer.transform(self.preprocessed_data['message'])
```

File: app.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    def rate_user_input(self):
+        self.rating = self.model.predict(self.vectorized)
```
Comment (Bilski311) [inline app.py:36]:
```suggestion
        self.rating = self.model.predict(self.vectorized_inputs)
```

**Review by Bilski311 — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (wdebsqi) [reply app.py:13]: Good idea.

File: app.py (hunk 2 above)
Comment (wdebsqi) [reply app.py:21]: It's necessary because we need to clean the array every time we get the new input.

File: app.py (hunk 3 above)
Comment (wdebsqi) [reply app.py:8]: Ok.

File: app.py (hunk 4 above)
Comment (wdebsqi) [reply app.py:22]: Ok.

File: app.py (hunk 2 above)
Comment (wdebsqi) [reply app.py:21]: Actually we don't need this in constructor. I'll remove it.

**Pull request merged**